### PR TITLE
Upgrade/rails5.2

### LINKED ISSUE #1083: Upgrade to Rails 5.2
Since we're making good headway on progress, let's see if we can get to Rails 5.2 to use the integrated features to support S3 upload.

### PR DESCRIPTION
**Pull request by yachtcaptain23**
Resolves #1083 

Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave-intl/publishers/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Tagged reviewers.
- [ ] Integrated piwik/matomo (for code that adds new buttons).
- [ ] Addressed or ignored all brakeman warnings

Test Plan:

Reviewer Checklist:

Tests
- [ ] Adequate test coverage exists to prevent regressions

Security:
- [ ] No raw SQL -- Always prefer ActiveRecord query helpers ([more info on StackOverflow](https://stackoverflow.com/questions/41410752/rails-5-sql-injection#41452695))
- [ ] XSS is mitigated -- Avoid `html_safe` and `raw`; escape untrusted content from users and 3rd party APIs ([Rails XSS guide](https://brakemanpro.com/2017/09/08/cross-site-scripting-in-rails) and [OWASP XSS Prevention Cheat Sheet](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet))

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -112,14 +118,14 @@ gem "u2f", "~> 1.0"
 # One-time passwords for 2fa
 gem "rotp", "~> 3.3"
 
-gem 'webpacker', '~> 3.2'
-
-# YouTube API client
-gem 'yt'
```
Comment (nvonpentz) [inline Gemfile]: I think we need the `yt` gem

File: Gemfile [hunk 2]
```diff
@@ -1,6 +1,12 @@
 # frozen_string_literal: true
 source "https://rubygems.org"
 
+gem 'actionmailer', '~> 5.2.0'
+gem 'actionpack', '~> 5.2.0'
+gem 'activemodel', '~> 5.2.0'
+gem 'activerecord', '~> 5.2.0'
+gem 'activesupport', '~> 5.2.0'
+
```
Comment (nvonpentz) [inline Gemfile]: Don't these automatically come with rails?

File: Procfile.dev [hunk 3]
```diff
@@ -1,2 +1,3 @@
 web: bundle exec puma -C config/puma.rb -e ${RACK_ENV:-development}
-worker: bundle exec sidekiq -C config/sidekiq.yml -e ${RACK_ENV:-development}
\ No newline at end of file
+worker: bundle exec sidekiq -C config/sidekiq.yml -e ${RACK_ENV:-development}
+webpacker: ./bin/webpack-dev-server
```
Comment (nvonpentz) [inline Procfile.dev]: Do we need to add this to the production procfile?

Do you know why we need to specify this when in the past we didn't?

File: app/models/channel.rb [hunk 4]
```diff
@@ -168,7 +168,7 @@ def update_last_verification_timestamp
   def should_register_channel_for_promo
     promo_running = Rails.application.secrets[:active_promo_id].present?  # Could use PromosHelper#active_promo_id
     publisher_enabled_promo = self.publisher.promo_enabled_2018q1?
-    promo_running && publisher_enabled_promo && verified_changed? && verified
+    promo_running && publisher_enabled_promo && saved_change_to_verified? && verified
```
Comment (nvonpentz) [inline app/models/channel.rb]: Is this change related to rails upgrade?

File: config/environments/development.rb [hunk 5]
```diff
@@ -21,7 +21,7 @@
   # Enable/disable caching. By default caching is disabled.
   if Rails.application.secrets[:redis_url]
     config.action_controller.perform_caching = true
-    config.cache_store = :redis_store
+    config.cache_store = :redis_cache_store, { url: Rails.application.secrets[:redis_url] }
```
Comment (nvonpentz) [inline config/environments/development.rb]: What does this change do?

File: app/models/channel.rb (hunk 4 above)
Comment (yachtcaptain23) [reply app/models/channel.rb]: Yes, it's related to https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Dirty.html

From what I understand, `changed?` was confusing because at-a-glance, it could be a change to the unsaved version or to the saved version.

File: config/environments/development.rb (hunk 5 above)
Comment (yachtcaptain23) [reply config/environments/development.rb]: https://www.engineyard.com/blog/rails-5-2-redis-cache-store

TL;DR Rails comes bundled with their own version of https://github.com/redis-store/redis-rails

File: Procfile.dev (hunk 3 above)
Comment (yachtcaptain23) [reply Procfile.dev]: Not sure if we need to add to the production Procfile. Webpacker detects changes to the javascript code so you don't need to refresh the page: https://webpack.js.org/concepts/hot-module-replacement/

However, since the assets are compiled during build time (and assets shouldn't change) I don't believe we're supposed to include webpacker. I'm not 100% confident though, so I'll test against staging-2.

File: Gemfile (hunk 2 above)
Comment (yachtcaptain23) [reply Gemfile]: They do, but upgrading dependencies via `bundler` wasn't elegant.

I'll remove these lines, and I think we should be fine as long as these changes are remarked in `Gemfile.lock`.
